test/utils/test_knowledge_utils.py

```diff
@@ -0,0 +1,357 @@
+#!/usr/bin/env python3
+
+"""
+Test script for world knowledge utilities
+"""
+
+import os
+import numpy as np
+
+from pyrobosim.core import WorldYamlLoader
+from pyrobosim.core import Robot
+from pyrobosim.core import World
+
+# import functions to test
+from pyrobosim.utils.knowledge import (
+    apply_resolution_strategy,
+    query_to_entity,
+    resolve_to_location,
+    resolve_to_object,
+)
+
+from pyrobosim.utils.pose import Pose
+from pyrobosim.utils.general import get_data_folder
+
+
+def load_world():
+    """Load a test world."""
+    world_file = os.path.join(get_data_folder(), "test_world.yaml")
+    return WorldYamlLoader().from_yaml(world_file)
+
+
+def test_apply_resolution_strategy():
+    # First, test all strategies with empty entity_list
+    entity_list = []
+    entity = apply_resolution_strategy([], entity_list, "first")
+    assert entity is None
+    entity = apply_resolution_strategy([], entity_list, "random")
+    assert entity is None
+    entity = apply_resolution_strategy([], entity_list, "nearest")
+    assert entity is None
+
+    # Test non-existent strategy
+    entity_list = ["don't return this"]
+    # catch the warning as an error
+    import warnings
+
+    warnings.filterwarnings("error")
+    try:
+        entity = apply_resolution_strategy([], entity_list, "non-existent")
+    except UserWarning:
+        # test passed, this should warn
+        pass
+    else:
+        assert (
+            False
+        ), "apply_resolution_strategy didn't warn on non-implemented resolution strategy"
+
+    # Test 'first' strategy
+    entity_list = ["First", "Second"]
+    entity = apply_resolution_strategy([], entity_list, "first")
+    assert entity == "First"
+
+    entity_list = [["list"], "second"]
+    entity = apply_resolution_strategy([], entity_list, "first")
+    assert entity == ["list"]
```

Comment:
I don't know if a nested list like this is a realistic scenario -- I'd get rid of it.